File: Go/01.Go-Basic-February-2022/10.advancedLoops-Exercise/06.cake.go
package main

import (
	"fmt"
	"math"
	"strconv"
)

func main() {
	var width, lenght int
	fmt.Scanln(&width)
	fmt.Scanln(&lenght)

	var pieces int = width * lenght

	var input string
	fmt.Scanln(&input)

	for input != "STOP" {
		takenPieces, _ := strconv.Atoi(input)
		pieces -= takenPieces

		if pieces <= 0 {
			fmt.Printf("No more cake left! You need %.0f pieces more.", math.Abs(float64(pieces)))
			break
		}

		fmt.Scanln(&input)
	}

	if input == "STOP" {
		fmt.Printf("%d pieces are left.", pieces)
	}

}

File: Python/01.Python-Basic-April-2019/09.while_loop_exercise/06.cake.py
length = int(input())
width = int(input())

cake_size = length * width

is_over = False
count = input()

while not count == 'STOP':
    count = int(count)
    cake_size -= count

    if cake_size <= 0:
        is_over = True
        print(f"No more cake left! You need {abs(cake_size)} pieces more.")
        break

    count = input()

if not is_over:
    print(f"{cake_size} pieces are left.")
